x/lockup/keeper/lock.go

```diff
@@ -493,6 +493,41 @@ func (k Keeper) beginForceUnlock(ctx sdk.Context, lock types.PeriodLock, coins s
 	return nil
 }
 
+func (k Keeper) BeginForceUnlockWithEndTime(ctx sdk.Context, lockID uint64, endTime time.Time) error {
+	lock, err := k.GetLockByID(ctx, lockID)
+	if err != nil {
+		return err
+	}
+	return k.beginForceUnlockWithEndTime(ctx, *lock, endTime)
+}
+
+func (k Keeper) beginForceUnlockWithEndTime(ctx sdk.Context, lock types.PeriodLock, endTime time.Time) error {
+	// remove lock refs from not unlocking queue if exists
+	err := k.deleteLockRefs(ctx, types.KeyPrefixNotUnlocking, lock)
```

Comment:
BeginForceUnlockWithEndTime is called when the lock is unlocking(in the UnpoolAllowedPool function if-then-else branch), but KeyPrefixNotUnlocking is used here

EDIT: no this is right because newLock is provided which is just created with notunlocking